app/scripts/modules/core/src/pipeline/config/stages/wercker/werckerStage.html

```diff
@@ -0,0 +1,188 @@
+         ng-if="!stage.master">(Select a master)</p>
+      <p class="form-control-static"
+         ng-if="viewState.masterIsParameterized || viewState.appIsParameterized">{{stage.app}}</p>
+      <div ng-if="stage.master && viewState.appsLoaded">
+        <ui-select class="form-control input-sm"
+                   ng-if="!viewState.masterIsParameterized && !viewState.appIsParameterized"
+                   ng-model="stage.app">
+          <ui-select-match placeholder="Select an Application...">{{$select.selected}}</ui-select-match>
+          <ui-select-choices repeat="app in apps | filter: $select.search"><span ng-bind-html="app | highlight: $select.search"></span></ui-select-choices>
+        </ui-select>
+      </div>
+      <div class="horizontal center" ng-if="stage.master && !viewState.appsLoaded">
+        <loading-spinner size="'small'"></loading-spinner>
+      </div>
+    </div>
+    <div class="col-md-1 text-center" ng-if="!viewState.masterIsParameterized && !viewState.appIsParameterized">
+      <a href
+         ng-click="werckerStageCtrl.refreshApps()"
+         tooltip-placement="right"
+         uib-tooltip="{{viewState.appsRefreshing ? 'Applications refreshing.' : 'Refresh application list' }}"
+        >
+              <span ng-class="{'fa-spin':viewState.appsRefreshing}"
+                    class="fa fa-sync-alt"></span>
+      </a>
+    </div>
+  </div>
+  
+  <div class="form-group">
+    <label class="col-md-2 col-md-offset-1 sm-label-right">Pipeline</label>
+    <div class="col-md-6">
+      <p class="form-control-static"
+         ng-if="!stage.master">(Select an Application)</p>
+      <p class="form-control-static"
+         ng-if="viewState.masterIsParameterized || viewState.jobIsParameterized">{{stage.job}}</p>
+      <div ng-if="stage.master && viewState.jobsLoaded">
+        <ui-select class="form-control input-sm"
+                   ng-if="!viewState.masterIsParameterized && !viewState.jobIsParameterized"
+                   ng-model="stage.pipeline">
+          <ui-select-match placeholder="Select a pipeline...">{{$select.selected}}</ui-select-match>
+          <ui-select-choices repeat="pipeline in pipelines | filter: $select.search"><span ng-bind-html="pipeline | highlight: $select.search"></span></ui-select-choices>
+        </ui-select>
+      </div>
+      <div class="horizontal center" ng-if="stage.master && !viewState.jobsLoaded">
+        <loading-spinner size="'small'"></loading-spinner>
+      </div>
+    </div>
+    <div class="col-md-1 text-center" ng-if="!viewState.masterIsParameterized && !viewState.jobIsParameterized">
+      <a href
+         ng-click="werckerStageCtrl.refreshJobs()"
+         tooltip-placement="right"
+         uib-tooltip="{{viewState.jobsRefreshing ? 'Jobs refreshing.' : 'Refresh job list' }}"
+        >
+              <span ng-class="{'fa-spin':viewState.jobsRefreshing}"
+                    class="fa fa-sync-alt"></span>
+      </a>
+    </div>
+  </div>
+ 
+<!--
```

Comment:
Ideally dead code should be removed rather than commented out.

Reply:
Did you mean rename all the the js file to ts? I got "Module not found: Error: Can't resolve './wercker/werckerTrigger.module.js' in '/Users/scchen/wrkspin/deck-ui/app/scripts/modules/core/src/pipeline/config/triggers'
Or which js file should I rename?

Reply:
Yes exactly - rename all the js files to ts.  Then in the places you've used `require()` you don't need to include `.js` in the filename.  This should fix the "Module not found" errors.